ironfish/src/network/peers/peerManager.ts

```diff
@@ -157,31 +157,26 @@ export class PeerManager {
    * Connect to a websocket by its uri. Establish a connection and solicit
    * the server's Identity.
    */
-  connectToWebSocketAddress(uri: string, isWhitelisted = false): Peer {
-    const url = parseUrl(uri)
-
-    if (!url.hostname) {
-      throw new Error(`Could not connect to ${uri} because hostname was not parseable`)
-    }
-
+  connectToWebSocketAddress(host: string, port: number, isWhitelisted = false): Peer {
     const peer = this.getOrCreatePeer(null)
-    peer.setWebSocketAddress(url.hostname, url.port)
+    peer.setWebSocketAddress(host, port)
     peer.isWhitelisted = isWhitelisted
 
     this.peerCandidates.addFromPeer(peer)
 
-    this.connectToWebSocket(peer)
+    if (this.canConnectToWebSocket(peer)) {
+      this.connectToWebSocket(peer)
+    }
+
     return peer
   }
 
   /**
-   * Connect to a peer using WebSockets
+   * Connect to a peer using WebSockets. This function forces a connection
+   * and doesn't check `canConnectToWebSocket`. Unless the connection needs to
+   * be forced `canConnectToWebSocket` should probably be called first
    * */
   connectToWebSocket(peer: Peer): boolean {
-    if (!this.canConnectToWebSocket(peer)) {
```

Comment:
I don't think we should move this out -- we only really want to force a connection in this one specific case we're adding, but this pattern causes you to have to remember to call `canConnectToWebSockets` in the other typical cases.

Plus, I think we might want more granularity around which checks we disable -- for example, it might make sense to not force the connection if the peer requested a disconnect, or if the peer already has an existing WebSocket connection.

Reply:
Agreed to pass down a parameter to `forceConnect`